modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/shibari_stand.dm

```diff
@@ -0,0 +1,270 @@
+	M.pixel_x = M.base_pixel_x + M.body_position_pixel_x_offset
+	M.pixel_y = M.base_pixel_y + M.body_position_pixel_y_offset - 4
+	M.layer = initial(M.layer)
+
+	cut_overlay(shibari_shadow_overlay)
+	cut_overlay(shibari_rope_overlay)
+	cut_overlay(shibari_rope_overlay_behind)
+
+	if(current_mob)
+		if(current_mob.handcuffed)
+			current_mob.handcuffed.dropped(current_mob)
+		current_mob.set_handcuffed(null)
+		current_mob.update_abstract_handcuffed()
+
+	if(ropee)
+		ropee.forceMove(get_turf(src))
+	current_mob = null
+
+/obj/item/restraints/handcuffs/milker/shibari
+	name = "ropes"
+	desc = "A shibari rope for restraining hands."
+	breakouttime = 2 MINUTES
+
+///////////////////////////
+//s-stand construction kit/
+///////////////////////////
+
+/obj/item/shibari_stand_kit
+	name = "shibari stand construction kit"
+	desc = "Construction requires a wrench."
+	icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_structures/bdsm_furniture.dmi'
+	throwforce = 0
+	icon_state = "shibari_kit"
+	w_class = WEIGHT_CLASS_HUGE
+	var/current_color = "pink"
+	var/color_changed = FALSE
+	var/static/list/shibarikit_designs
+
+/obj/item/shibari_stand_kit/proc/populate_shibarikit_designs()
+	shibarikit_designs = list(
+		"pink" = image (icon = src.icon, icon_state = "shibari_kit_pink"),
+		"teal" = image (icon = src.icon, icon_state = "shibari_kit_teal"),
+		"black" = image (icon = src.icon, icon_state = "shibari_kit_black"),
+		"red" = image (icon = src.icon, icon_state = "shibari_kit_red"),
+		"white" = image (icon = src.icon, icon_state = "shibari_kit_white"))
+
+// Default initialization
+/obj/item/shibari_stand_kit/Initialize()
+	. = ..()
+	populate_shibarikit_designs()
+	update_icon_state()
+	update_icon()
+
+/obj/item/shibari_stand_kit/update_icon_state()
+	. = ..()
+	icon_state = "[initial(icon_state)]_[current_color]"
+
+//to change model
+/obj/item/shibari_stand_kit/screwdriver_act(mob/living/user, obj/item/tool)
+	var/choice = show_radial_menu(user,src, shibarikit_designs, custom_check = CALLBACK(src, .proc/check_menu, user, tool), radius = 36, require_near = TRUE)
+	if(!choice)
+		return FALSE
+	if(tool.use_tool(src, user, 0, volume=40))
+		to_chat(user, span_notice("You switched the frame's plastic fittings color to [choice]."))
+		current_color = choice
+		update_appearance()
+	return TRUE
+
+/obj/item/shibari_stand_kit/proc/check_menu(mob/living/user)
+	if(!istype(user))
+		return FALSE
+	if(user.incapacitated())
+		return FALSE
+	return TRUE
+
+/obj/item/shibari_stand_kit/wrench_act(mob/living/user, obj/item/tool)
+	to_chat(user, span_notice("You begin fastening the frame to the floor."))
+	if(tool.use_tool(src, user, 8 SECONDS, volume=50))
+		to_chat(user, span_notice("You assemble the frame."))
+		var/obj/structure/chair/shibari_stand/C = new
+		C.icon_state = "shibari_stand_[current_color]"
+		C.forceMove(get_turf(src))
+		qdel(src)
+	return TRUE
+
+/obj/structure/chair/shibari_stand/wrench_act(mob/living/user, obj/item/I)
```

Comment:
descriptive vars

Reply:
No single-letter variables. Make it meaningful variable names.